.github/workflows/ci.yml

```diff
@@ -4,86 +4,48 @@ on:
-      - uses: wagoid/commitlint-github-action@v1
-        env:
-          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-      - name: if lint failed
-        if: ${{ failure() }}
-        uses: peter-evans/create-or-update-comment@v1
-        with:
-          issue-number: ${{ github.event.pull_request.number }}
-          body: |
-            Thanks for your contribution :heart:
-            :broken_heart: Unfortunately, this PR has one ore more **bad commit messages**, it can not be merged. To fix this problem, please refer to:
-            - [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
-            - [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
-
-            Note, other CI tests will *not* *start* until the commit messages get fixed.
-
-            This message will be deleted automatically when the commit messages get fixed.
-          reaction-type: "eyes"
+          python-version: 3.7
+      - name: Install requirements
+        run: pip install -r .github/requirements-test.txt
+      - name: Lint with flake8
+        run: flake8
 
-  lint-flake-8:
+  check-black:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python 3.7
         uses: actions/setup-python@v2
         with:
           python-version: 3.7
-      - name: Lint with flake8
-        run: |
-          pip install flake8
-          # stop the build if there are Python syntax errors or undefined names
-          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics --exclude .git,__pycache__,docs/source/conf.py,old,build,dist,tests/,jina/resources/
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude .git,__pycache__,docs/source/conf.py,old,build,dist,tests/,jina/resources/
+      - name: Install requirements
+        run: pip install -r .github/requirements-test.txt
+      - name: check black
+        run: black --check .
 
-  check-black:
+  check-isort:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
       - name: Set up Python 3.7
         uses: actions/setup-python@v2
         with:
           python-version: 3.7
-      - id: file_changes
-        uses: Ana06/get-changed-files@v1.2
-      - name: check black
-        run: ./scripts/black.sh
-        env:
-          CHANGED_FILES: ${{ steps.file_changes.outputs.added_modified }}
+      - name: Install requirements
+        run: pip install -r .github/requirements-test.txt
+      - name: check isort
+        run: isort --check .
 
   prep-testbed:
     runs-on: ubuntu-latest
-    needs: [commit-lint, lint-flake-8, check-black]
+    needs: [lint-flake-8, check-black, check-isort]
```

Comment:
i think a general better thing to do is to only apply changes on "updated files within the PR", not all files.

Reply:
If you start from a state where all checks pass, it is the same. Running time is negligible (<1s for all), so I don't see a good reason why we need it - it just introduces an extra step in CI